lib/framework/sync_job_runner.rb

```diff
@@ -55,14 +55,19 @@ def cron_parser(cronline)
     end
 
     def should_sync?
-      return false unless @connector_settings.scheduling_settings[:enabled]
-      return true if @connector_settings[:sync_now]
+      # sync_now should have priority over cron
+      return true if @connector_settings[:sync_now] == true
+      scheduling_settings = @connector_settings.scheduling_settings
+      return false unless scheduling_settings[:enabled] == true
 
-      last_synced = @connector_settings.scheduling_settings[:last_synced]
+      last_synced = scheduling_settings[:last_synced]
```

Comment:
I see, probably it was my mistake when porting the code and it should have been `self[:last_synced]`